packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,188 @@
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+[Amazon FSx](https://docs.aws.amazon.com/fsx/?id=docs_gateway) provides fully managed third-party file systems with the
+native compatibility and feature sets for workloads such as Microsoft Windows–based storage, high-performance computing,
+machine learning, and electronic design automation.
+
+Amazon FSx supports two file system types: [Lustre](https://docs.aws.amazon.com/fsx/latest/LustreGuide/index.html) and
+[Windows](https://docs.aws.amazon.com/fsx/latest/WindowsGuide/index.html) File Server.
+
+## FSx for Lustre
+
+Amazon FSx for Lustre makes it easy and cost-effective to launch and run the popular, high-performance Lustre file
+system. You use Lustre for workloads where speed matters, such as machine learning, high performance computing (HPC),
+video processing, and financial modeling.
+
+The open-source Lustre file system is designed for applications that require fast storage—where you want your storage
+to keep up with your compute. Lustre was built to solve the problem of quickly and cheaply processing the world's
+ever-growing datasets. It's a widely used file system designed for the fastest computers in the world. It provides
+submillisecond latencies, up to hundreds of GBps of throughput, and up to millions of IOPS. For more information on
+Lustre, see the [Lustre website](http://lustre.org/).
+
+As a fully managed service, Amazon FSx makes it easier for you to use Lustre for workloads where storage speed matters.
+Amazon FSx for Lustre eliminates the traditional complexity of setting up and managing Lustre file systems, enabling
+you to spin up and run a battle-tested high-performance file system in minutes. It also provides multiple deployment
+options so you can optimize cost for your needs.
+
+Amazon FSx for Lustre is POSIX-compliant, so you can use your current Linux-based applications without having to make
+any changes. Amazon FSx for Lustre provides a native file system interface and works as any file system does with your
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+const storageCapacity = 1200;
+const lustreConfiguration = {
+  deploymentType: FsxLustreDeploymentType.SCRATCH_2
+};
+
+const fileSystem = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  storageCapacity,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
```

Comment:
You should be taking a `SubnetSelection` object.

Must it always be exactly one subnet per filesystem? Does that mean you must pay cross-AZ traffic costs if you have a fleet of machines spread out across different AZs? 

EFS has a mechanism for creating "mount points" in different AZs.

Reply:
Yes, an FSx file system exists in a single AZ, so you would incur costs for cross-AZ traffic. FSx for Lustre can only exist in a single subnet and FSx for Windows Server can exist in a single subnet with a backup in a different subnet. There is no concept of mount points in FSx.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-fsx-filesystem.html#cfn-fsx-filesystem-subnetids

Should I still use the SubnetSelection object if the user can only select a single subnet? That interface doesn't have a good way of restricting the result to a single item.

Reply:
That is a fair point. Okay let's keep it this way then.